include/deal.II/multigrid/mg_transfer.templates.h

```diff
@@ -361,58 +389,271 @@ set_component_to_block_map (const std::vector<unsigned int> &map)
 
 /* --------- MGLevelGlobalTransfer<LinearAlgebra::distributed::Vector> ------- */
 
-template <typename Number>
-template <int dim, typename Number2, int spacedim>
-void
-MGLevelGlobalTransfer<LinearAlgebra::distributed::Vector<Number> >::copy_to_mg
-(const DoFHandler<dim,spacedim>                        &mg_dof_handler,
- MGLevelObject<LinearAlgebra::distributed::Vector<Number> > &dst,
- const LinearAlgebra::distributed::Vector<Number2>          &src) const
+/*
+ * internal functions for LA::d::Vector and LA::d::BlockVector
+ */
+namespace internal
 {
-  AssertIndexRange(dst.max_level(), mg_dof_handler.get_triangulation().n_global_levels());
-  AssertIndexRange(dst.min_level(), dst.max_level()+1);
-  reinit_vector(mg_dof_handler, component_to_block_map, dst);
+  template<typename Number>
+  inline
+  LinearAlgebra::distributed::Vector<Number> &get_block(LinearAlgebra::distributed::Vector<Number> &vec, const unsigned int )
```

Comment:
i should combine these with those in `MatrixFreeOperators`, probably move to a separate header.